### PR TITLE
v1.11 backports 2022-01-20

### PR DESCRIPTION
**Pull request by kkourt**
v1.11 backports 2022-01-20

 * #17988 -- contrib: Fix backport submission for own PRs (@joestringer)
 * #17980 -- builder: Enable building images on arm64 (@jrajahalme)
      * last commit update cilium-{runtime,builder}, was redone following: https://docs.cilium.io/en/stable/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images
 * #18024 -- doc: getting started minor fixes (@kaworu)
 * #18468 -- docs: Don't mark pre-upgrade step as "recommended" (@pchaigno)
 * #18471 -- helm: Update links in values.yaml (@sayboras)
 * #18506 -- Fix EncryptStatusSuite.TestCountUniqueIPsecKeys (@tklauser)
 * #18482 -- Allow `install-no-conntrack-iptables-rules` when masquerading is disabled (@pchaigno)
      * There was a conflict (seemed minor to me) in this PR, resolved by only adding the new `MasqueradingEnabled()` function.

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 17988 17980 18024 18468 18471 18506 18482; do contrib/backporting/set-labels.py $pr done 1.11; done
```

#### Note
Skipped  #18483: since it contained a mistake which is fixed by #18582, so they should probably be backported together.

### CODE REVIEW

**Review by nbusseneau — APPROVED** [review]

File: .github/workflows/build-images-ci.yaml [hunk 1]
```diff
@@ -109,7 +109,7 @@ jobs:
             quay.io/${{ github.repository_owner }}/${{ matrix.name }}-ci:v1.11-race
             quay.io/${{ github.repository_owner }}/${{ matrix.name }}-ci:${{ steps.tag.outputs.tag }}-race
           build-args: |
-            BASE_IMAGE=quay.io/cilium/cilium-runtime:3e2ea3c996d8372ba65ee14058a716f797c67496@sha256:f5ea135fce72807158e520d1754e826b138cf9e97adde528fd992b3e79ef6ee8
+            BASE_IMAGE=quay.io/cilium/cilium-runtime:805506ff0dc0a9886fc1a8ce53191275f8baf62e
```
Comment (nbusseneau) [inline .github/workflows/build-images-ci.yaml]: Wrong click sorry => I think there might be an issue here with the image reference as we're missing the `sha256` pin.

**Review by nbusseneau — DISMISSED** [review]

File: .github/workflows/build-images-ci.yaml (hunk 1 above)
Comment (kkourt) [reply .github/workflows/build-images-ci.yaml]: I'm following steps here: https://docs.cilium.io/en/stable/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

I think they images are build in the build deployment and running the scripts will add their checksums.

**Comment by nbusseneau** [comment]
re: #18506 -- #18500 is not failing on v1.11 because we did not update the VM images there. Is it necessary to backport it?
cc @tklauser

**Review by pchaigno — APPROVED** [review]
My changes look good :pray:

**Comment by tklauser** [comment]
> re: #18506 -- #18500 is not failing on v1.11 because we did not update the VM images there. Is it necessary to backport it? cc @tklauser
> 
> EDIT: I guess we'll see soon enough when CI runs on this PR if this causes issues.

I think it's still worth to backport because 78a4afd fixes a potential issue with the test itself (regardless of the kernel update) and the netlink library bump 59eedd2 will be necessary not only for passing said test but also to be able to run ipsec on newer kernels.

**Comment by kkourt** [comment]
/test-backport-1.11

**Review by joestringer — APPROVED** [review]

**Review by sayboras — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Comment by kkourt** [comment]
/test-1.20-4.9

Hit #16847: https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/1210/testReport/junit/Suite-k8s-1/20/K8sPolicyTest_Multi_node_policy_test_with_L7_policy_using_connectivity_check_to_check_datapath/

**Comment by kkourt** [comment]
/test-1.20-4.9

**Comment by kkourt** [comment]
/test-1.22-4.19

**Review by jrajahalme — APPROVED** [review]
Looking good on my part :-)

**Comment by kkourt** [comment]
CI is green, and got all approvals from original authors. Marking `ready-to-merge`.

**Pull request merged**